pkg/boot/bzimage/kver.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package bzimage
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+)
+
+/*
+values from kernel documentation and libmagic src
+
+off val
+510 0xAA55
+514 HdrS
+526	(4 bytes) != 0x0000
+526 (2 bytes, little endian) + 0x200 -> start of null-terminated version string
+*/
+
+const kverMax = 1024 //arbitrary
+
+var (
+	ErrBootSig = errors.New("missing 0x55AA boot sig")
+	ErrBadSig  = errors.New("missing kernel header sig")
+	ErrBadOff  = errors.New("null version string offset")
+	ErrParse   = errors.New("parse error")
+)
+
+//Read kernel version string. See also: (*BZImage)Kver()
+func KVer(k io.ReadSeeker) (string, error) {
+	var buf = make([]byte, kverMax)
+	_, err := k.Seek(0, io.SeekStart)
+	if err != nil {
+		return "", err
+	}
+	_, err = k.Read(buf[:530])
+	if err != nil {
+		return "", err
+	}
+	if !bytes.Equal(buf[510:512], []byte{0x55, 0xaa}) {
+		return "", ErrBootSig
+	}
+	if string(buf[514:518]) != "HdrS" {
+		return "", ErrBadSig
+	}
+	if bytes.Equal(buf[526:530], []byte{0, 0, 0, 0}) {
+		return "", ErrBadOff
+	}
+	off := int64(binary.LittleEndian.Uint16(buf[526:528])) + 0x200
+	_, err = k.Seek(off, io.SeekStart)
+	if err != nil {
+		return "", err
+	}
+	if _, err := k.Read(buf[:]); err != nil {
+		return "", err
+	}
+	return nullterm(buf), nil
+}
+
+//Read kernel version string. See also: KVer()
```

Comment:
I'd suggest you run golint and go vet on this, and it will then complain about some of your function comments :-)

// KVer returns a version for a bzImage

Also is an error possible here? Should it return (string, error)?

Reply:
I fixed these functions and a number of other things in the package that `lint` was unhappy with. These were all comments except for one variable with an underscore.

Reply:
I also changed this function to return (string, error)